packages/core/src/payment/strategies/bnz/bnz-payment-strategy.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { merge } from 'lodash';
+import { of } from 'rxjs';
+
+import { createCheckoutStore, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { HostedFormFactory } from '../../../hosted-form';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { getOrderRequestBody } from '../../../order/internal-orders.mock';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import { getCybersource } from '../../payment-methods.mock';
+import { CardinalThreeDSecureFlowV2 } from '../cardinal';
+import { CreditCardPaymentStrategy } from '../credit-card';
+
+import BNZPaymentStrategy from './bnz-payment-strategy';
+
+describe('BankOfNewZealandPaymentStrategy', () => {
+    let hostedFormFactory: HostedFormFactory;
+    let orderActionCreator: Pick<OrderActionCreator, 'submitOrder'>;
+    let paymentActionCreator: Pick<PaymentActionCreator, 'submitPayment'>;
+    let paymentMethod: PaymentMethod;
+    let state: InternalCheckoutSelectors;
+    let store: CheckoutStore;
+    let strategy: BNZPaymentStrategy;
+    let threeDSecureFlow: Pick<CardinalThreeDSecureFlowV2, 'prepare' | 'start'>;
+
+    beforeEach(() => {
+        paymentMethod = {
+            ...getCybersource(),
+            clientToken: 'foo',
+        };
+
+        store = createCheckoutStore();
+
+        orderActionCreator = {
+            submitOrder: jest.fn(() => of()),
+        };
+
+        paymentActionCreator = {
+            submitPayment: jest.fn(() => of()),
+        };
+
+        hostedFormFactory = {} as HostedFormFactory;
+
+        threeDSecureFlow = {
+            prepare: jest.fn(() => Promise.resolve()),
+            start: jest.fn(() => Promise.resolve()),
+        };
+
+        state = store.getState();
+
+        jest.spyOn(store, 'dispatch')
+            .mockResolvedValue(state);
+
+        jest.spyOn(store, 'getState')
+            .mockReturnValue(state);
+
+        jest.spyOn(state.paymentMethods, 'getPaymentMethodOrThrow')
+            .mockReturnValue(paymentMethod);
+
+        strategy = new BNZPaymentStrategy(
+            store,
+            orderActionCreator as OrderActionCreator,
+            paymentActionCreator as PaymentActionCreator,
+            hostedFormFactory,
+            threeDSecureFlow as CardinalThreeDSecureFlowV2
+        );
+    });
+
+    it('is special type of credit card strategy', () => {
+        expect(strategy)
+            .toBeInstanceOf(CreditCardPaymentStrategy);
+    });
+
+    describe('#initialize', () => {
+        it('throws error if payment method is not defined', async () => {
```

Comment:
Usually we describe the unit test like this. It is related to the unit test conventions
```suggestion
        it('throws error when payment method is not defined', async () => {
```